src/controllers/admin/sounds.js

```diff
@@ -1,24 +1,46 @@
 'use strict';
 
-var meta = require('../../meta');
+var plugins = require('../../plugins');
+var db = require('../../database');
 
 var soundsController = {};
 
 soundsController.get = function (req, res, next) {
-	meta.sounds.getFiles(function (err, sounds) {
+	db.getObject('settings:sounds', function (err, settings) {
 		if (err) {
 			return next(err);
 		}
+		
+		settings = settings || {};
 
-		sounds = Object.keys(sounds).map(function (name) {
-			return {
-				name: name
-			};
-		});
+		var types = [
+			'notification',
+			'chat-incoming',
+			'chat-outgoing',
+		];
+		var output = {};
+
+		types.forEach(function (type) {
+			var soundpacks = plugins.soundpacks.map(function (pack) {
+				var sounds = Object.keys(pack.sounds).map(function (soundName) {
+					var value = pack.name + ' | ' + soundName;
+					return {
+						name: soundName,
+						value: value,
+						selected: value === settings[type],
+					};
+				});
 
-		res.render('admin/general/sounds', {
-			sounds: sounds
+				return {
+					name: pack.name,
+					sounds: sounds,
+				};
+			});
+
+			output[type.replace('-', '_') + '_sound'] = soundpacks;
```

Comment:
This looks like a hack... why does the sound name need to be underscore spaced, but is provided in `types` with dashed spacing?